src/maintenance.cpp

```diff
@@ -4,35 +4,255 @@
+            break;
+          case 4:
+            ctx->setMaintenanceActiveModule(ActiveModule::DOLIBARR);
+            break;
+          case 0:
+          default:
+            ctx->setMaintenanceActiveModule(ActiveModule::NONE);
+        }
+
+        clearScreen();
+        M5.Lcd.println("Maintenance Mode for: ");
+        M5.Lcd.println(ACTIVE_MODULES[range]);
+        LOG_DEBUG("[MAINT.] Changing module: %s\n", ACTIVE_MODULES[range]);
+        range = 1;
+      }
+
+      if (buttons.BtnC->pressedFor(HOLD_BUTTON_TIME)) {
+        LOG_DEBUG("[MODE] Exit maintenance mode \n");
+        ctx->requestOperationModeChange(OperationMode::CONFIGURATION);
+      }
+    }
+  } while (interruptibleTaskPauseMs(BUTTONS_READ_INTERVAL));
+}
+
+static void runConveyor(TaskContext *ctx) {
+  Buttons &buttons = ctx->getHardware()->buttons;
+  Conveyor &conveyor = ctx->getHardware()->conveyor;
+  do {
+    if (ctx->getCurrentMaintenanceModule() == ActiveModule::CONVEYOR) {
+      buttons.update();
+      conveyor.update();
+      if (buttons.BtnA->wasPressed()) {
+        LOG_DEBUG("[CONV.] RUN CONVEYOR \n");
+        M5.Lcd.println("RUN MOTOR");
+        conveyor.start();
+      } else if (buttons.BtnC->wasPressed()) {
+        M5.Lcd.println("STOP MOTOR");
+        conveyor.stop();
+      }
+
+      if (buttons.BtnB->pressedFor(HOLD_BUTTON_TIME)) {
+        LOG_DEBUG("[MAINT.] Exit module %s\n", ACTIVE_MODULES[1]);
+        ctx->setMaintenanceActiveModule(ActiveModule::NONE);
+        clearScreen();
+        showMenu();
+      }
+    }
+  } while (interruptibleTaskPauseMs(BUTTONS_READ_INTERVAL));
+}
 
-[[deprecated("TEMPORARY CODE: replace with actual maintenance routines")]]
-static void temporaryFakeMaintenanceModeTask(TaskContext *ctx) {
+static void startSorter(TaskContext *ctx) {
   Buttons &buttons = ctx->getHardware()->buttons;
+  Sorter &sorter = ctx->getHardware()->sorter;
 
+  int angle = 50;
+  int realPosition = 0;
   do {
-    buttons.update();
-    if (buttons.BtnB->wasPressed()) {
-      LOG_DEBUG("[BTN] B pressed\n");
-      ctx->requestOperationModeChange(OperationMode::CONFIGURATION);
+    if (ctx->getCurrentMaintenanceModule() == ActiveModule::SORTER) {
+      buttons.update();
+      if (buttons.BtnC->wasPressed()) {
+        LOG_INFO("[SORT.] INCREASE ANGLE \n");
+        angle += 1;
+
+        clearScreen();
+        M5.Lcd.println("ANGLE : ");
+        M5.Lcd.print(angle);
+      } else if (buttons.BtnA->wasPressed()) {
+        LOG_INFO("[SORT.] DECREASE ANGLE \n");
+        angle -= 1;
+        clearScreen();
+        M5.Lcd.println("ANGLE : ");
+        M5.Lcd.print(angle);
+
+      } else if (buttons.BtnB->wasPressed()) {
+        M5.Lcd.println("MOVE");
+        if (realPosition < angle) {
+          while (realPosition < angle) {
+            sorter.moveWithSpecificAngle(realPosition);
+            realPosition += 1;
+            interruptibleTaskPauseMs(CHANGE_ANGLE_DELAY);
```

Comment:
You need to check the return value of `interruptibleTaskPauseMs` (which returns a `bool`) :
* When `false` you need to break out of the while loop(s) and exit the whole task function.